enterprise/server/auth/auth.go

```diff
@@ -647,7 +649,15 @@ func ClaimsFromSubID(env environment.Env, ctx context.Context, subID string) (*C
 	if err != nil {
 		return nil, err
 	}
-	return userClaims(u), nil
+	eg := ""
+	if c := requestcontext.ProtoRequestContextFromContext(ctx); c != nil && c.GetGroupId() != "" {
+		for _, g := range u.Groups {
+			if g.Group.GroupID == c.GetGroupId() {
+				eg = c.GetGroupId()
+			}
+		}
+	}
+	return userClaims(u, eg), nil
```

Comment:
Should we update `authutil.EffectiveGroupID` to only consult u.GroupID instead of looking at the request context? (Just for the sake of making `u.GroupID` the source of truth for the preferred, authenticated group ID). Seems like it should be safe since we don't return user claims anywhere except for this function

Reply:
Removed `authutil.EffectiveGroup` altogether

Reply:
sg